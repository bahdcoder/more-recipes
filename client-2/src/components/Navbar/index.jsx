import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="navbar navbar-expand-lg navbar-custom">
    <Link className="navbar-brand" to="/">
      <img src={`${process.env.PUBLIC_URL}/img/logo.png`} className="navbar-logo-img mr-2" alt=''/>
      More-recipes
    </Link>
    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <span className="navbar-toggler-icon" />
    </button>
    <div className="collapse navbar-collapse" id="navbarSupportedContent">
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <a href="" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></a>
        </li>
        <li className="nav-item mr-3">
          <a className="nav-link" href="">Recipes <span className="sr-only">(current)</span></a>
        </li>
        <form className="form-inline my-2 my-lg-0">
          <input className="form-control mr-sm-2" placeholder="Search" aria-label="Search" type="text" />
        </form>
      </ul>
    </div>
  </nav>
);

export default Navbar;
