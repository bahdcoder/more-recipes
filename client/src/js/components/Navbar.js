import React from 'react';
import { NavLink } from 'react-router-dom';

export default class NavBar extends React.Component {
  render() {
    return (
      <nav className="navbar navbar-expand-lg navbar-custom">
        <NavLink className="navbar-brand" to="/">
          <img src="../../../assets/img/logo.png" className="navbar-logo-img mr-2" alt />
          More-recipes
        </NavLink>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <NavLink to="/recipes/create" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></NavLink>
            </li>
            <li className="nav-item mr-3">
              <NavLink to="/recipes" className="nav-link">Recipes <span className="sr-only">(current)</span></NavLink>
            </li>
            <form className="form-inline my-2 my-lg-0">
              <input className="form-control mr-sm-2" type="text" placeholder="Search" aria-label="Search" />
            </form>
          </ul>
        </div>
      </nav>
    );
  }
}