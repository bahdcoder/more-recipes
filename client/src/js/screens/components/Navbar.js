import React from 'react';
import { Link } from 'react-router';
import Gravatar from 'react-gravatar';

export default class NavBar extends React.Component {
  render() {
    let authUser = this.props.authUser;
    let navbarUser = '';
    if (authUser) {
      navbarUser = (
        <li className="nav-item dropdown">
          <a className="nav-link dropdown-toggle" href="javascript:;" id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
          <Gravatar className="mr-2 navbar-avatar-img" email={this.props.authUser.user.email}/>
                  Hey, {this.props.authUser.user.name}
          </a>
          <div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
              <a className="dropdown-item" href="profile.html">My profile</a>
              <a className="dropdown-item" href="favorites.html">My favorites</a>
              <a className="dropdown-item" href="settings.html">Notification settings</a>
              <a className="dropdown-item" 
                 href="javascript:;" 
                 onClick={async () => {
                   await this.props.signOut();
                   this.props.router.push('/');
                 }}>Sign out</a>
          </div>
      </li>
      );
    }
    return (
      <nav className="navbar navbar-expand-lg navbar-custom">
        <Link className="navbar-brand" to="/">
          <img src="../../../assets/img/logo.png" className="navbar-logo-img mr-2" />
          More-recipes
        </Link>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link to="/recipes/create" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></Link>
            </li>
            <li className="nav-item mr-3">
              <Link to="/recipes" className="nav-link">Recipes <span className="sr-only">(current)</span></Link>
            </li>
            <form className="form-inline my-2 my-lg-0">
              <input className="form-control mr-sm-2" type="text" placeholder="Search" aria-label="Search" />
            </form>
            {navbarUser}
          </ul>
        </div>
      </nav>
    );
  }
}