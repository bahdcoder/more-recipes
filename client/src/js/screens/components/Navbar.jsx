/* eslint-disable no-script-url */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import Gravatar from 'react-gravatar';

import logo from './../../../assets/img/logo.png';

/**
 * The navbar component
 * @param {object} props
 * @returns {jsx} jsx
 */
const Navbar = (props) => {
  const { authUser } = props;
  let navbarUser;
  if (authUser) {
    navbarUser = (
      <li className="nav-item dropdown">
        <a className="nav-link dropdown-toggle" id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
          <Gravatar className="mr-2 navbar-avatar-img" email={props.authUser.user.email} />
          Hey, {props.authUser.user.name}
        </a>
        <div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
          <Link className="dropdown-item" to={`/user/${props.authUser.user.id}`}>My profile</Link>
          <Link id="myRecipes" className="dropdown-item" to={`/user/${props.authUser.user.id}/recipes`}>My recipes</Link>
          <Link className="dropdown-item" to="/my/favorites">My favorites</Link>
          <a
            className="dropdown-item logout"
            id="logout"
            onClick={async () => {
              await props.signOut();
              props.router.push('/');
            }}
          >
            Sign out
          </a>
        </div>
      </li>
    );
  }

  return (
    <nav className="navbar navbar-expand-lg navbar-custom">
      <Link className="navbar-brand" to="/">
        <img src={logo} className="navbar-logo-img mr-2" alt="" />
        More-recipes
      </Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link to="/recipes/create" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></Link>
          </li>
          <li className="nav-item mr-3">
            <Link to="/recipes" className="nav-link">Recipes <span className="sr-only">(current)</span></Link>
          </li>
          <form
            className="form-inline my-2 my-lg-0"
            onSubmit={(event) => {
              event.preventDefault();
              if (props.location.pathname === '/recipes') {
                props.triggerGetRecipesCatalog();
              } else {
                props.router.push(`/recipes?query=${props.location.query.query || ''}`);
              }
            }}
          >
            <input
              className="form-control mr-sm-2"
              value={props.location.query.query || ''}
              onChange={(event) => {
                props.changeRouterQueryParams('query', event.target.value, props.location);
              }}
              type="text"
              placeholder="Search recipes"
              aria-label="Search"
            />
          </form>
          {navbarUser}
        </ul>
      </div>
    </nav>
  );
};

Navbar.propTypes = {
  authUser: PropTypes.objectOf(PropTypes.any),
  signOut: PropTypes.func.isRequired,
  location: PropTypes.objectOf(PropTypes.any).isRequired,
  triggerGetRecipesCatalog: PropTypes.func.isRequired,
  changeRouterQueryParams: PropTypes.func.isRequired,
  router: PropTypes.objectOf(PropTypes.any).isRequired
};

Navbar.defaultProps = {
  authUser: null
};

export default Navbar;
