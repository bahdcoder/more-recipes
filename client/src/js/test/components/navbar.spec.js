/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow, mount } from 'enzyme';

import setupTests from '../setupTests';
import Navbar from '../../screens/components/Navbar';

setupTests();

describe('The Navbar component', () => {
  const signedOutProps = {
    authUser: null,
    triggerGetRecipesCatalog() { },
    changeRouterQueryParams() { },
    router: {},
    signOut() { },
    location: { query: { query: '' }, pathname: '' }
  };
  const signedInProps = {
    authUser: {
      access_token: '',
      user: {
        id: 1,
        name: 'Test User',
        email: 'test@user.com'
      }
    },
    triggerGetRecipesCatalog: () => {},
    changeRouterQueryParams() { },
    router: { push() {} },
    signOut() { },
    location: { query: { query: '' }, pathname: '/recipes' }
  };
  test('should mount without crashing', () => {
    const tree = shallow((
      <div>
        <Navbar {...signedInProps} />
      </div>
    ));
    expect(tree).toMatchSnapshot();
  });
  test('should display user information if logged in', () => {
    const wrapper = mount(<Navbar {...signedInProps} />);
    expect(wrapper.find('.dropdown-menu').length).toBe(1);
  });
  test('should not display user information if logged in', () => {
    const wrapper = mount(<Navbar {...signedOutProps} />);
    expect(wrapper.find('.dropdown-menu').length).toBe(0);
  });
  test.skip('should call triggerGetRecipesCatalog function when search recipes form is submitted', () => {
    const wrapper = mount(<Navbar {...signedInProps} />);
    const spy = jest.spyOn(signedInProps, 'triggerGetRecipesCatalog');
    wrapper.find('form').simulate('submit');
    expect(spy).toHaveBeenCalled();
  });
});
