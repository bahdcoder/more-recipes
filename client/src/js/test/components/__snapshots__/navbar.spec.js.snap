// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navbar component should mount without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <Navbar
      authUser={
        Object {
          "access_token": "",
          "user": Object {
            "email": "test@user.com",
            "id": 1,
            "name": "Test User",
          },
        }
      }
      changeRouterQueryParams={[Function]}
      location={
        Object {
          "pathname": "/recipes",
          "query": Object {
            "query": "",
          },
        }
      }
      router={
        Object {
          "push": [Function],
        }
      }
      signOut={[Function]}
      triggerGetRecipesCatalog={[Function]}
    />
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Navbar
        authUser={
          Object {
            "access_token": "",
            "user": Object {
              "email": "test@user.com",
              "id": 1,
              "name": "Test User",
            },
          }
        }
        changeRouterQueryParams={[Function]}
        location={
          Object {
            "pathname": "/recipes",
            "query": Object {
              "query": "",
            },
          }
        }
        router={
          Object {
            "push": [Function],
          }
        }
        signOut={[Function]}
        triggerGetRecipesCatalog={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "authUser": Object {
          "access_token": "",
          "user": Object {
            "email": "test@user.com",
            "id": 1,
            "name": "Test User",
          },
        },
        "changeRouterQueryParams": [Function],
        "location": Object {
          "pathname": "/recipes",
          "query": Object {
            "query": "",
          },
        },
        "router": Object {
          "push": [Function],
        },
        "signOut": [Function],
        "triggerGetRecipesCatalog": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Navbar
          authUser={
            Object {
              "access_token": "",
              "user": Object {
                "email": "test@user.com",
                "id": 1,
                "name": "Test User",
              },
            }
          }
          changeRouterQueryParams={[Function]}
          location={
            Object {
              "pathname": "/recipes",
              "query": Object {
                "query": "",
              },
            }
          }
          router={
            Object {
              "push": [Function],
            }
          }
          signOut={[Function]}
          triggerGetRecipesCatalog={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authUser": Object {
            "access_token": "",
            "user": Object {
              "email": "test@user.com",
              "id": 1,
              "name": "Test User",
            },
          },
          "changeRouterQueryParams": [Function],
          "location": Object {
            "pathname": "/recipes",
            "query": Object {
              "query": "",
            },
          },
          "router": Object {
            "push": [Function],
          },
          "signOut": [Function],
          "triggerGetRecipesCatalog": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
